### PR TITLE
feat(bedrockagent): add agent, prompt, datasource and knowledge base

### LINKED ISSUE #425: [resource] Bedrock agent resources
Some Bedrock resources are in a dedicated API called "Bedrock Agent", this is currently not covered by the deletion of Bedrock resources I am working on

Documentation: https://docs.aws.amazon.com/bedrock/latest/userguide/agents-how.html

Update: PR created 😄

### PR DESCRIPTION
**Pull request by npellegrin**
Closes #425

### CODE REVIEW

**Comment by npellegrin** [comment]
New resources:

 - BedrockAgent
 - BedrockDataSource
 - BedrockKnowledgeBase
 - BedrockPrompt

Scan with feature activated

```
us-east-1 - BedrockAgent - AQSNLDAAXY - [ID: "AQSNLDAAXY", Name: "agent-quick-start-6j13n", Status: "NOT_PREPARED"] - would remove
us-east-1 - BedrockKnowledgeBase - OHVCRR2RER - [ID: "OHVCRR2RER", Name: "knowledge-base-quick-start-ntzce", Status: "ACTIVE"] - would remove
us-east-1 - BedrockPrompt - ENA3KYO939 - [ID: "ENA3KYO939", Name: "test", Version: "DRAFT"] - would remove
us-east-1 - BedrockDataSource - FTVCXNEISG - [ID: "FTVCXNEISG", KnowledgeBaseID: "OHVCRR2RER", Name: "knowledge-base-quick-start-jbqrq-data-source", Status: "AVAILABLE"] - would remove
```

Nuke with feature activated

```
us-east-1 - BedrockAgent - AQSNLDAAXY - [ID: "AQSNLDAAXY", Name: "agent-quick-start-6j13n", Status: "NOT_PREPARED"] - triggered remove
us-east-1 - BedrockKnowledgeBase - OHVCRR2RER - [ID: "OHVCRR2RER", Name: "knowledge-base-quick-start-ntzce", Status: "ACTIVE"] - triggered remove
us-east-1 - BedrockPrompt - ENA3KYO939 - [ID: "ENA3KYO939", Name: "test", Version: "DRAFT"] - triggered remove
us-east-1 - BedrockDataSource - FTVCXNEISG - [ID: "FTVCXNEISG", KnowledgeBaseID: "OHVCRR2RER", Name: "knowledge-base-quick-start-jbqrq-data-source", Status: "AVAILABLE"] - triggered remove

Removal requested: 3 waiting, 1 failed, 0 skipped, 0 finished

us-east-1 - BedrockAgent - AQSNLDAAXY - [ID: "AQSNLDAAXY", Name: "agent-quick-start-6j13n", Status: "NOT_PREPARED"] - removed
us-east-1 - BedrockKnowledgeBase - OHVCRR2RER - [ID: "OHVCRR2RER", Name: "knowledge-base-quick-start-ntzce", Status: "ACTIVE"] - removed
us-east-1 - BedrockPrompt - ENA3KYO939 - [ID: "ENA3KYO939", Name: "test", Version: "DRAFT"] - removed
us-east-1 - BedrockDataSource - FTVCXNEISG - [ID: "FTVCXNEISG", KnowledgeBaseID: "OHVCRR2RER", Name: "knowledge-base-quick-start-jbqrq-data-source", Status: "AVAILABLE"] - removed

Removal requested: 0 waiting, 0 failed, 0 skipped, 4 finished
```

File: resources/bedrock-agent-datasource.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+func (l *BedrockDataSourceLister) List(_ context.Context, o interface{}) ([]resource.Resource, error) {
+	opts := o.(*nuke.ListerOpts)
+
+	svc := bedrockagent.New(opts.Session)
+	resources := make([]resource.Resource, 0)
+
+	// Require to have workspaces identifiers to query components
+	knowledgeBaseListResponse, err := ListKnowledgeBaseForDataSource(svc)
+
```
Comment (ekristen) [inline resources/bedrock-agent-datasource.go:36]: Please remove empty line.

**Review by ekristen — CHANGES_REQUESTED** [review]

File: resources/bedrock-agent-datasource.go (hunk 1 above)
Comment (npellegrin) [reply resources/bedrock-agent-datasource.go]: Done 👍

**Review by ekristen — APPROVED** [review]

**Pull request merged**